src/v/redpanda/application.cc

```diff
@@ -557,9 +591,29 @@ void application::start() {
       _log.info, "Started Kafka API server listening at {}", conf.kafka_api());
 
     vlog(_log.info, "Successfully started Redpanda!");
+    _health_ctx
+      .invoke_on_all([](health_context& ctx) {
+          ctx.started = true;
```

Comment:
so it seems like we want to start-up the http server as early as possible and have it respond with live = true. then, readiness is more squishy. we could for example wait until the initial recovery has finished and the kafka server has started?

i think having these three health check endpoints is enough for this patch, and we can customize how these flags get set during the start-up life cycle? that seems like it is at least partially a co-design with the k8s pod stuff?

Reply:
In hindsight, I think everything will work as intended without any of these probes.

Reply:
neato.